src/ring_theory/power_series/basic.lean

```diff
@@ -969,6 +971,45 @@ lemma is_unit_constant_coeff (φ : power_series R) (h : is_unit φ) :
   is_unit (constant_coeff R φ) :=
 mv_power_series.is_unit_constant_coeff φ h
 
+open finset nat
+
+variables {A : Type*} [comm_ring A]
+
+/-- The ring homomorphism taking a power series `f(X)` to `f(aX)`. -/
+noncomputable def eval_mul_hom (a : A) : power_series A →+* power_series A :=
+{ to_fun :=   λ f, power_series.mk $ λ n, a^n * (power_series.coeff A n f),
+  map_zero' := by { ext, simp only [linear_map.map_zero, power_series.coeff_mk, mul_zero], },
+  map_one' := by { ext1, simp only [mul_boole, power_series.coeff_mk, power_series.coeff_one],
+                split_ifs, { rw [h, pow_zero], }, refl, },
+  map_add' := by {intros, ext, norm_num, rw mul_add, },
+  map_mul' := by {intros, ext, rw [power_series.coeff_mul, power_series.coeff_mk,
+              power_series.coeff_mul, finset.mul_sum], apply sum_congr rfl, norm_num,
+              intros b c H, rw [<-H, pow_add], ring, }, }
```

Comment:
You can relax this definition substantially to:
```suggestion
/-- The ring homomorphism taking a power series `f(X)` to `f(a • X)`. -/
noncomputable def eval_smul_hom {S A} [comm_semiring S] [semiring A] [algebra S A] (a : S) :
  power_series A →+* power_series A :=
{ to_fun :=   λ f, power_series.mk $ λ n, a^n • power_series.coeff A n f,
  map_zero' := by { ext, simp only [linear_map.map_zero, power_series.coeff_mk, smul_zero], },
  map_one' := by { ext1, simp only [mul_boole, power_series.coeff_mk, power_series.coeff_one],
                split_ifs, { rw [h, pow_zero, one_smul], }, rw smul_zero, },
  map_add' := by {intros, ext, norm_num, },
  map_mul' := by {intros, ext, rw [power_series.coeff_mul, power_series.coeff_mk,
              power_series.coeff_mul, finset.smul_sum], apply sum_congr rfl, norm_num,
              intros b c H, rw [<-H, pow_add, mul_smul], },}

/-- The ring homomorphism taking a power series `f(X)` to `f(a * X)`. -/
noncomputable def eval_mul_hom {R} [comm_semiring R] (a : R) : power_series R →+* power_series R :=
eval_smul_hom a
```
which takes A from `comm_ring` to `semiring`. You'll also have to change
```diff
-instance [comm_ring R]       : algebra R       (power_series R) := by apply_instance
+instance {A} [semiring A] [comm_semiring R] [algebra R A] :
+  algebra R (power_series A) := by apply_instance
```
around line 800 to make this work.

Reply:
(note that I care about this because it means I can evaluate `exp` on the quaternions!)

Reply:
Actually, this is a mess - lets not do this in this PR